pup-0003.md

```diff
@@ -0,0 +1,157 @@
+---
+title: Git Workflow
+author: David Davis
+created: 01-Apr-2017
+status: Accepted
+---
+
+## Summary
+
+This PUP is to redefine our git workflow to use cherry-picking instead of
+merging commits forward from devel branches. Using this model, PRs would be
+made against master and then bug fixes would be cherry-picked to the devel
+branch as needed. Hotfixes would be cherry-picked from release branches into
+devel branches and master.
+
+## Motivation
+
+Currently in Pulp, we merge features to master. However, bug fixes are merged
+to devel branches which then are merged to master (a process called "merging
+forward"). More info about merging forward can be found in our documentation:
+
+http://docs.pulpproject.org/en/2.12/dev-guide/contributing/merging.html#merging-your-changes
+
+Also, a good illustration of this process is outlined in the following webpage:
+
+http://nvie.com/posts/a-successful-git-branching-model/
+
+As a result, our git history is complex with multiple branches feeding into one
+another and numerous merge commits. Using cherry-picks would simplify our git
+history by keeping it more linear and eliminating merge commits.
+
+Opening a PR requires knowledge of our current process. This can create a
+barrier to entry for community members. With the cherry-pick model, all PRs are
+opened against master and the onus to update older dev branches is on the person
+merging (who usually has better knowledge of which branches a commit
+needs to be applied to).
+
+We've also seen some serious mistakes where branches are accidentally merged
+into other branches.  These situations have occurred multiple times and each
```

Comment:
The OCD Editor in me is bothered that this paragraph uses two spaces between sentences. ;)